Makefile

```
@@ -52,34 +60,26 @@ docker-push:
 # Rules for verification, formatting, linting, testing and cleaning #
 #####################################################################
 
-.PHONY: revendor
-revendor:
-	@GO111MODULE=on go mod vendor
-	@GO111MODULE=on go mod tidy
-	@chmod +x $(REPO_ROOT)/vendor/github.com/gardener/gardener/hack/*
-	@chmod +x $(REPO_ROOT)/vendor/github.com/gardener/gardener/hack/.ci/*
-	@$(REPO_ROOT)/hack/update-github-templates.sh
-
 .PHONY: clean
 clean:
 	@$(shell find ./example -type f -name "controller-registration.yaml" -exec rm '{}' \;)
-	@$(REPO_ROOT)/vendor/github.com/gardener/gardener/hack/clean.sh ./cmd/... ./pkg/...
+	@bash $(GARDENER_HACK_DIR)/clean.sh ./cmd/... ./pkg/...
 
 .PHONY: check-generate
 check-generate:
 	@$(REPO_ROOT)/vendor/github.com/gardener/gardener/hack/check-generate.sh $(REPO_ROOT)
 
 .PHONY: generate
-generate: $(HELM)
-	@$(REPO_ROOT)/vendor/github.com/gardener/gardener/hack/generate-sequential.sh ./charts/... ./cmd/... ./pkg/...
+generate: $(VGOPATH) $(HELM) $(YQ)
+	echo $(shell git describe --abbrev=0 --tags) > VERSION
+	@REPO_ROOT=$(REPO_ROOT) VGOPATH=$(VGOPATH) GARDENER_HACK_DIR=$(GARDENER_HACK_DIR) bash $(GARDENER_HACK_DIR)/generate-sequential.sh ./charts/... ./cmd/... ./pkg/...
 
 .PHONY: generate-in-docker
-generate-in-docker: revendor $(HELM) $(YQ)
-	echo $(shell git describe --abbrev=0 --tags) > VERSION
+generate-in-docker: tidy $(HELM) $(YQ)
 	docker run --rm -i$(DOCKER_TTY_ARG) -v $(PWD):/go/src/github.com/metal-stack/gardener-extension-audit golang:$(GO_VERSION) \
 		sh -c "cd /go/src/github.com/metal-stack/gardener-extension-audit \
+				&& chown -R $(shell id -u):$(shell id -g) ." \
```

I think there is a problem with a double quote. And maybe it's also good to require `install` before to make sure we always fetch the latest tooling with every update.

```suggestion
generate-in-docker: tidy install $(HELM) $(YQ)
	docker run --rm -i$(DOCKER_TTY_ARG) -v $(PWD):/go/src/github.com/metal-stack/gardener-extension-audit golang:$(GO_VERSION) \
		sh -c "cd /go/src/github.com/metal-stack/gardener-extension-audit \
				&& chown -R $(shell id -u):$(shell id -g) . \
```

---

Fixed `generate-in-docker`. 

Issue with hint: [here](https://stackoverflow.com/questions/73485958/how-to-correct-git-reporting-detected-dubious-ownership-in-repository-withou)

Problem:
- make generate within docker-container does have other permissions than .git folder introduces 
- failes due to `dubiose ownership` 
- VERSION is empty -> can not create controller-registration.yaml
- to bypass this problem for one git command: `git -c safe.directory=<DIR> <YOUR-SUB-COMMAND>`